epochStart/bootstrap/bootStrapShardProcessor.go

```diff
@@ -0,0 +1,409 @@
+}
+
+func (e *bootStrapShardProcessor) createRequestHandler() (process.RequestHandler, error) {
+	requestersContainerArgs := requesterscontainer.FactoryArgs{
+		RequesterConfig:                 e.generalConfig.Requesters,
+		ShardCoordinator:                e.shardCoordinator,
+		MainMessenger:                   e.mainMessenger,
+		FullArchiveMessenger:            e.fullArchiveMessenger,
+		Marshaller:                      e.coreComponentsHolder.InternalMarshalizer(),
+		Uint64ByteSliceConverter:        uint64ByteSlice.NewBigEndianConverter(),
+		OutputAntifloodHandler:          disabled.NewAntiFloodHandler(),
+		CurrentNetworkEpochProvider:     disabled.NewCurrentNetworkEpochProviderHandler(),
+		MainPreferredPeersHolder:        disabled.NewPreferredPeersHolder(),
+		FullArchivePreferredPeersHolder: disabled.NewPreferredPeersHolder(),
+		PeersRatingHandler:              disabled.NewDisabledPeersRatingHandler(),
+		SizeCheckDelta:                  0,
+	}
+	requestersFactory, err := requesterscontainer.NewMetaRequestersContainerFactory(requestersContainerArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	container, err := requestersFactory.Create()
+	if err != nil {
+		return nil, err
+	}
+
+	err = requestersFactory.AddShardTrieNodeRequesters(container)
+	if err != nil {
+		return nil, err
+	}
+
+	finder, err := containers.NewRequestersFinder(container, e.shardCoordinator)
+	if err != nil {
+		return nil, err
+	}
+
+	requestedItemsHandler := cache.NewTimeCache(timeBetweenRequests)
+	return requestHandlers.NewResolverRequestHandler(
+		finder,
+		requestedItemsHandler,
+		e.whiteListHandler,
+		maxToRequest,
+		core.MetachainShardId,
+		timeBetweenRequests,
+	)
+}
+
+func (e *bootStrapShardProcessor) createResolversContainer() error {
+	dataPacker, err := partitioning.NewSimpleDataPacker(e.coreComponentsHolder.InternalMarshalizer())
+	if err != nil {
+		return err
+	}
+
+	storageService := disabled.NewChainStorer()
+
+	payloadValidator, err := validator.NewPeerAuthenticationPayloadValidator(e.generalConfig.HeartbeatV2.HeartbeatExpiryTimespanInSec)
+	if err != nil {
+		return err
+	}
+
+	// TODO - create a dedicated request handler to be used when fetching required data with the correct shard coordinator
+	//  this one should only be used before determining the correct shard where the node should reside
+	log.Debug("epochStartBootstrap.createRequestHandler", "shard", e.shardCoordinator.SelfId())
+	resolversContainerArgs := resolverscontainer.FactoryArgs{
+		ShardCoordinator:                    e.shardCoordinator,
+		MainMessenger:                       e.mainMessenger,
+		FullArchiveMessenger:                e.fullArchiveMessenger,
+		Store:                               storageService,
+		Marshalizer:                         e.coreComponentsHolder.InternalMarshalizer(),
+		DataPools:                           e.dataPool,
+		Uint64ByteSliceConverter:            uint64ByteSlice.NewBigEndianConverter(),
+		NumConcurrentResolvingJobs:          10,
```

Comment:
we need these numbers from a config. maybe add a TODO.

Reply:
Right, that's a very good point.
This is the code from master, so we should fix it there. Will add a TODO, nevertheless